packages/jest-types/__typetests__/jest.test.ts

```diff
@@ -98,6 +98,9 @@ expectError(jest.enableAutomock('moduleName'));
 expectType<typeof jest>(jest.isolateModules(() => {}));
 expectError(jest.isolateModules());
 
+expectType<typeof Promise>(jest.isolateModulesAsync(() => {}));
```

Comment:
Does it allow the callback to be async?

Reply:
Yes it does. That's pretty much the reason why I set out to do this. Should the signature be different?

Reply:
I see, should be a `Promise.resolve(jest.fn())`

Reply:
Return type changed to `Promise<void>` in 5810116f02db4f5b0ca78a0f49be40de0f925a1d